### PR TITLE
Automigration: Improve the viewport/backgrounds automigration

### LINKED ISSUE #32605: [Bug]: addon-global-api automigration does not fully transform the code
### Describe the bug

Here are a few issues from QA, some might be edge case, but a few aren't.

### Viewports

**1- The properties `defaultOrientation` and `disabled` ([docs for reference](https://storybook.js.org/docs/8/essentials/viewport#parameters)) aren't transformed. Here's an example of before/after.** 

before:
```ts
export const Mobile: Story = {
  parameters: {
    viewport: {
      defaultOrientation: 'portrait',
      defaultViewport: 'iphonex',
      disable: true,
    },
  },
}
```

after:
```ts
export const Mobile: Story = {
  parameters: {
    viewport: {
      defaultOrientation: 'portrait',
      disable: true
    },
  },

  globals: {
    viewport: {
      value: "iphonex",
      isRotated: false
    }
  }
}
```

**2-  const defined viewport parameter does not get transformed**
```ts
const viewport = {
  defaultViewport: 'iphonex',
}

export const Mobile: Story = {
  parameters: {
    viewport
  },
}
```

**3- importing a value from viewports addon to a story does not get transformed**
```ts
import { MINIMAL_VIEWPORTS } from 'storybook/viewport'

export const Mobile: Story = {
  parameters: {
    viewport: {
      defaultViewport: MINIMAL_VIEWPORTS.mobile2
    },
  },
}
```

### Backgrounds

**1. The property `disable` is not considered**

before:
```ts
export const Desktop: Story = {
  parameters: {
    backgrounds: {
      // 👇 Set default background value for all component stories
      default: 'Gray',
      disable: true
    },
  },
}
```

after:
```ts
export const Desktop: Story = {
  parameters: {
    backgrounds: {
      disabled: true
    },
  },

  globals: {
    backgrounds: {
      value: "gray"
    }
  }
}
```

**2. The property `value` is not considered**

before:
```ts
export const Mobile: Story = {
  parameters: {
    backgrounds: {
      default: 'Light',
      values: [
        { name: 'Gray', value: '#CCC' },
      ],
    },
  },
}
```

after:
```ts
export const Mobile: Story = {
  parameters: {
    backgrounds: {
      values: [
        { name: 'Gray', value: '#CCC' },
      ]
    }
  },

  globals: {
    backgrounds: {
      value: "light"
    }
  }
}
```

**3. The property `grid` is not considered**

This story does not get transformed
```ts
export const Default: Story = {
  parameters: {
    backgrounds: {
      grid: {
        cellSize: 20,
        opacity: 0.5,
        cellAmount: 5,
        offsetX: 16, // Default is 0 if story has 'fullscreen' layout, 16 if layout is 'padded'
        offsetY: 16, // Default is 0 if story has 'fullscreen' layout, 16 if layout is 'padded'
      },
    },
  },
}
```

### Reproduction link

-

### Reproduction steps

_No response_

### System

```bash
-
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Closes https://github.com/storybookjs/storybook/issues/32605

## What I did

I fixed the first and the third issue, which were reported. The second one is too complex to support it and is also considered rather edge-casey. The rest of the problems were wrongly reported, because the outcome of the transformations is actually okay!

#### The changes in this PR are covered in the following automated tests:

- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

### 🦋 Canary release

This pull request has been released as version `0.0.0-pr-32619-sha-7a4ae9e3`. Try it out in a new sandbox by running `npx storybook@0.0.0-pr-32619-sha-7a4ae9e3 sandbox` or in an existing project with `npx storybook@0.0.0-pr-32619-sha-7a4ae9e3 upgrade`.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-32619-sha-7a4ae9e3`](https://npmjs.com/package/storybook/v/0.0.0-pr-32619-sha-7a4ae9e3) |
| **Triggered by** | @valentinpalkovic |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`valentin/improve-viewport-background-automigration`](https://github.com/storybookjs/storybook/tree/valentin/improve-viewport-background-automigration) |
| **Commit** | [`7a4ae9e3`](https://github.com/storybookjs/storybook/commit/7a4ae9e3ad3f42df53995c5af70a5015445a6728) |
| **Datetime** | Wed Oct  8 09:32:16 UTC 2025 (`1759915936`) |
| **Workflow run** | [18340285885](https://github.com/storybookjs/storybook/actions/runs/18340285885) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/publish.yml) or locally with `gh workflow run --repo storybookjs/storybook publish.yml --field pr=32619`_
</details>

## Summary by CodeRabbit

- New Features
  - Improved automigration for viewports and backgrounds: supports literal and referenced defaults, converts backgrounds array to options, maps defaultOrientation to rotation, renames disable→disabled, and consolidates defaults into globals.

- Bug Fixes
  - Ensures migrated stories accurately preserve viewport/background defaults and rotation without breaking references.

- Tests
  - Updated test harness and snapshots to exercise the new migration pipeline and output format.

### CODE REVIEW

File: code/lib/cli-storybook/src/automigrate/fixes/addon-globals-api.ts [hunk 1]
```diff
@@ -310,18 +290,38 @@ function transformStoryFile(
...
+          }
+
           newGlobals.properties.push(
             t.objectProperty(
               t.identifier('viewport'),
               t.objectExpression([
-                t.objectProperty(t.identifier('value'), t.stringLiteral(defaultViewport.value)),
-                t.objectProperty(t.identifier('isRotated'), t.booleanLiteral(false)),
+                t.objectProperty(t.identifier('value'), viewportValue),
+                t.objectProperty(t.identifier('isRotated'), t.booleanLiteral(isRotated)),
```
Comment (yannbf) [inline code/lib/cli-storybook/src/automigrate/fixes/addon-globals-api.ts:324]: if there was no defaultOrientation ever defined, does it make sense to add `isRotated` to every story?

**Review by yannbf — APPROVED** [review]
